report/report.go

```diff
@@ -1,24 +1,61 @@
 package report
 
 import (
+	"encoding/json"
 	"html/template"
 	"log"
 	"os"
+	"path/filepath"
+	"strings"
 	"talisman/detector"
 )
 
+const htmlFileName string = "talisman_reports/report.html"
+const jsonFileName string = "talisman_reports/report.json"
 // GenerateReport generates a talisman scan report in html format
-func GenerateReport(r *detector.DetectionResults) {
+func GenerateReport(r *detector.DetectionResults, directory string) string {
+
+	var path string
+	var htmlFilePath strings.Builder
+	var jsonFilePath strings.Builder
+	if directory == "" {
+		path = "talisman_reports"
+		htmlFilePath.WriteString(htmlFileName)
+		jsonFilePath.WriteString(jsonFileName)
+	} else  {
+		path = filepath.Join(directory, "/talisman_reports")
+		htmlFilePath.WriteString(directory)
+		htmlFilePath.WriteString("/")
```

Comment:
Please use filepath.Join(directory, htmlFileName) instead of WriteString.
Also omit using '/' in the argument values of filepath.Join

Reply:
Have made the necessary modifications